src/main.c

```diff
@@ -2408,15 +2409,58 @@ void handle_mouse_input() {
     }
 }
 
-void handle_movement(double dt) {
+/// [issue](https://github.com/Team-10-But-Better/Craft/issues/9)
+/// This function returns the movement speed based on whether the character should teleport.
+///
+///\param teleport : whether the character should teleport
+///\param teleportHeldLastFrame : whether the teleport button was previously held
+///\param groundedMovementSpeed : default movement speed
+
+int getGroundedMovementSpeed(bool teleport, bool *teleportHeldLastFrame, int groundedMovementSpeed)
+{
+    if (teleport)
+    {
+        if (*teleportHeldLastFrame == false){
+            *teleportHeldLastFrame = true;
+            return 600;
+        }
+        *teleportHeldLastFrame = true;
+    }
+    else
+    {
+        *teleportHeldLastFrame = false;
+    }
+
+    return groundedMovementSpeed;
+}
+
+void handle_movement(double dt, bool *teleportHeldLastFrame)
+{
     static float dy = 0;
     State *s = &g->players->state;
     int sz = 0;
     int sx = 0;
+    char shouldTeleport = 0;
+    int groundedMovementSpeed = 5;
+
     if (!g->typing) {
         float m = dt * 1.0;
         g->ortho = glfwGetKey(g->window, CRAFT_KEY_ORTHO) ? 64 : 0;
         g->fov = glfwGetKey(g->window, CRAFT_KEY_ZOOM) ? 15 : 65;
+        groundedMovementSpeed = getGroundedMovementSpeed(glfwGetKey(g->window, CRAFT_KEY_TELEPORT), teleportHeldLastFrame, groundedMovementSpeed);
+       /* if (glfwGetKey(g->window, CRAFT_KEY_TELEPORT))
```

Comment:
Please don't forget to remove commented out code to keep the code clean. The more commented out code their is, the harder the program is to analyze, especially for future developers who may not know why said code is commented out and whether it is okay to remove.